tests/iface_namingmode/test_iface_namingmode.py

```diff
@@ -0,0 +1,829 @@
+import logging
+import pytest
+
+from common.devices import AnsibleHostBase
+from common.utilities import wait
+from netaddr import IPAddress
+
+logger = logging.getLogger(__name__)
+
+@pytest.fixture(scope='module', autouse=True)
+def setup(duthost):
+    """
+    Sets up all the parameters needed for the interface naming mode tests
+
+    Args:
+        duthost: AnsiblecHost instance for DUT
+    Yields:
+        setup_info: dictionary containing port alias mappings, list of
+        working interfaces, minigraph facts
+    """
+    hwsku = duthost.facts['hwsku']
+    minigraph_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+    port_alias_facts = duthost.port_alias(hwsku=hwsku)['ansible_facts']
+    up_ports = minigraph_facts['minigraph_ports'].keys()
+    default_interfaces = port_alias_facts['port_name_map'].keys()
+
+    port_alias = list()
+    port_name_map = dict()
+    port_alias_map = dict()
+    port_speed = dict()
+
+    # Change port alias names to make it common for all platforms
+    logger.info('Updating common port alias names in redis db')
+    for i, item in enumerate(default_interfaces):
+        port_alias_new = 'TestAlias{}'.format(i)
+        port_alias_old = port_alias_facts['port_name_map'][item]
+        port_alias.append(port_alias_new)
+        port_name_map[item] = port_alias_new
+        port_alias_map[port_alias_new] = item
+        port_speed[port_alias_new] = port_alias_facts['port_speed'][port_alias_old]
+
+        # Update port alias name in redis db
+        duthost.command('redis-cli -n 4 HSET "PORT|{}" alias {}'.format(item, port_alias_new))
+
+    upport_alias_list = [ port_name_map[item] for item in up_ports ]
+    setup_info = {
+         'default_interfaces' : default_interfaces,
+         'minigraph_facts' : minigraph_facts,
+         'port_alias' : port_alias,
+         'port_name_map' : port_name_map,
+         'port_alias_map' : port_alias_map,
+         'port_speed' : port_speed,
+         'up_ports' : up_ports,
+         'upport_alias_list' : upport_alias_list
+    }
+
+    yield setup_info
+
+    logger.info('Reverting the port alias name in redis db to the actual values')
+    for item in default_interfaces:
+        port_alias_old = port_alias_facts['port_name_map'][item]
+        duthost.command('redis-cli -n 4 HSET "PORT|{}" alias {}'.format(item, port_alias_old))
```

Comment:
The original ansible version of this testing reboot the DUT to recover after the testing is done. Is setting port alias to old names enough to recover or do we need to perform a reboot or config reload here?

Reply:
All the configurations changes done as part of the testing are reverted by the script itself. So reboot or config reload is not required.